### PR TITLE
hack pkt and support port_metadata_unpack

### PR DESCRIPTION
**Pull request by RabbitWhite1**
A temporal hack for pkt to support dummy port_metadata_unpack.

### CODE REVIEW

**Review by fruffy — CHANGES_REQUESTED** [review]
There is a better way to solve this. Either mark the packet with an appropriate qualifier (in/inout/out) [here](https://github.com/fruffy/flay/blob/54ac630052d39d0fe18708c2e017372dc25b646b/targets/tofino/test/p4include/tofino1_base.p4#L539C10-L539C30) or ignore any parameter that has a Type_Extern type.

**Comment by RabbitWhite1** [comment]
Changing to in/inout/out requires specifying a type, which I don't know if there is a suitable one for packet_in parameter. So I think the latter one is better. But instead of ignoring all Type_Extern parameters, what about just ignoring the parameter that is of Type_Extern type and the type name is `packet_in` for now?

**Comment by fruffy** [comment]
> Changing to in/inout/out requires specifying a type, which I don't know if there is a suitable one for packet_in parameter. So I think the latter one is better. But instead of ignoring all Type_Extern parameters, what about just ignoring the parameter that is of Type_Extern type and the type name is `packet_in` for now?

I would just not resolve any `Type_Extern`. There is no clear semantics to how you would even do that and we can still access the type in the lambda later.

**Comment by RabbitWhite1** [comment]
I see. That makes sense!

File: core/expression_resolver.cpp [hunk 1]
```diff
@@ -292,10 +292,17 @@ bool ExpressionResolver::preorder(const IR::MethodCallExpression *call) {
     for (size_t idx = 0; idx < callArguments->size(); ++idx) {
         const auto *arg = callArguments->at(idx);
         const auto *param = methodParams.at(idx);
-        const auto *computedExpr = computeResult(arg->expression);
-        if (param->direction == IR::Direction::InOut || param->direction == IR::Direction::Out) {
-            auto stateVar = ToolsVariables::convertReference(arg->expression);
-            computedExpr = new IR::InOutReference(stateVar, computedExpr);
+        const IR::Expression *computedExpr;
+        if (param->type->is<IR::Type_Extern>()) {
+            // Parameters of `Type_Extern` will be parsed (if necessary) in lambda hanlders.
```
Comment (fruffy) [inline core/expression_resolver.cpp:297]:
```suggestion
            // Parameters of `Type_Extern` will be parsed (if necessary) in the lambda handlers.
```

**Review by fruffy — APPROVED** [review]

**Pull request merged**